### PR TITLE
Fix initialCapacity=0 in PriorityQueue serializers

### LINKED ISSUE #16224: Incorrect deserialization in PriorityQueueStreamSerializer
There is incorrect deserialization in `PriorityQueueStreamSerializer` and `PriorityBlockingQueueStreamSerializer`. The same issue was fixed in Jet by https://github.com/hazelcast/hazelcast-jet/blob/3.1-maintenance/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/serialization/PriorityQueueHook.java#L62

### PR DESCRIPTION
**Pull request by viliam-durina**
In `PriorityQueue` and `PriorityBlockingQueue` the initial capacity must be 1
or more.

Fixes #16224

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractCollectionStreamSerializer.java [hunk 1]
```diff
@@ -26,17 +26,15 @@
...
 
     @Override
     public void write(ObjectDataOutput out, CollectionType collection) throws IOException {
         int size = collection.size();
         out.writeInt(size);
-        if (size > 0) {
-            for (Object o : collection) {
-                out.writeObject(o);
-            }
+        for (Object o : collection) {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractCollectionStreamSerializer.java]: What is the reason to get rid of the if check? In theory scalar replacement could be applied to the iterator; preventing it to be created on the heap. But I'm not sure how well the JIT deal with this; especially with JDK 8.

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/PriorityBlockingQueueStreamSerializer.java [hunk 2]
```diff
@@ -43,11 +43,8 @@ public void write(ObjectDataOutput out, PriorityBlockingQueue<E> collection) thr
     @Override
     public PriorityBlockingQueue<E> read(ObjectDataInput in) throws IOException {
         Comparator<E> comparator = in.readObject();
-
         int size = in.readInt();
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/PriorityBlockingQueueStreamSerializer.java]: How can it be that the value 0 was received?

Comment (viliam-durina) [reply hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/PriorityBlockingQueueStreamSerializer.java]: If an empty priorityQueue was saved.

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractCollectionStreamSerializer.java (hunk 1 above)
Comment (viliam-durina) [reply hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractCollectionStreamSerializer.java]: I consider it to be a bad style. Googled a bit and found this claim for OpenJDK Hotspo 1.8.0_121:
>Practically speaking, simple iterators will usually be stack allocatable if you just use them to iterate.

Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractCollectionStreamSerializer.java]: Good to know.

**Pull request merged**